### ISSUE TITLE
100% like sails hook ?

### ISSUE THREAD

**Issue by jaumard** [trailsjs/trailpack-auth#2]
Did you plan to make it work like @tjwebb sails hook ? I mean it's create all files directly on sails project.
Personally I don't want this on Trails version. I prefer take advantage of classes and only override models/controllers/policies if needed. Can you explain more how you want this trailpack to work ? 

Same for `trailpack-permission`. Maybe I will help with this two if you want

**Comment by tjwebb** [comment]
You're right, we should definitely do this "trails-style" using inheritance instead of code generation + mixins.

Probably most of the code will need to be re-written, but that's ok. We can be somewhat patient, and prioritize good design and testing. One big difference is that I'd like to support JWT first. Currently sails-auth primarily focuses on using cookies to maintain sessions.

One thing we'll want to think about is the ability to standardize the Policy API (https://github.com/trailsjs/trails/issues/89) since we'll need to publish a set of those that works in all the webservers. We can continue the discussion in https://github.com/trailsjs/trails/issues/89 about what we want this to look like.

**Comment by s-devaney** [comment]
+1 for JWT support

**Comment by jaumard** [comment]
@tjwebb I make a hook for sails that use Passport https://github.com/jaumard/sails-hook-passport like your sails-auth but I don't use generate file. I think some code can be use for this trailpack. 
You lock this repo (and permissions) we can't fork it for now. It's something you want to do personally ?

**Comment by tjwebb** [comment]
@jaumard which repo is locked? I don't think I've locked anything

**Comment by jaumard** [comment]
@tjwebb `trailpack-auth` and `trailpack-permissions` are empty repo so we can't fork them to work on them :)
